File: dsa/recursion/PrimeFactor.java
public class PrimeFactor {
    public static void main(String args[]) {
        int a = 45;
        System.out.println("Prime factors of " + a + " are :");
        pf(a);
    }

    static void pf(int n) {
        if(n==1){
            return;
        }
        int temp=n;
        for(int i=2;i<=Math.sqrt(n);i++){
            if(n%i==0){
                temp = i;
                break;
            }
        }
        System.out.println(temp);
        pf((int)(n/temp));
    }
}

File: python/primeFactor.py
def factor(a):
    if(a==1):
        return 1
    i=2
    while(a%i!=0):
        i=i+1
    return i
# print(factor(1))
def primeFactor(n):
    if(n==1):
        return
    f=factor(n)
    print(f,end=" ")
    primeFactor(n/f)
primeFactor(77)
print()
